client/src/components/menubar/index.js

```diff
@@ -0,0 +1,693 @@
+      graphInteractionMode
+    } = this.props;
+    const { pendingClipPercentiles } = this.state;
+
+    const haveBothCellSets =
+      !!differential.celllist1 && !!differential.celllist2;
+
+    const clipMin =
+      pendingClipPercentiles?.clipPercentileMin ?? clipPercentileMin;
+    const clipMax =
+      pendingClipPercentiles?.clipPercentileMax ?? clipPercentileMax;
+    const activeClipClass =
+      clipPercentileMin > 0 || clipPercentileMax < 100
+        ? " bp3-intent-warning"
+        : "";
+
+    // constants used to create selection tool button
+    let selectionTooltip;
+    let selectionButtonClass;
+    if (selectionTool === "brush") {
+      selectionTooltip = "Brush selection";
+      selectionButtonClass = "bp3-icon-select";
+    } else {
+      selectionTooltip = "Lasso selection";
+      selectionButtonClass = "bp3-icon-polygon-filter";
+    }
+
+    return (
+      <div
+        style={{
+          paddingTop: 10,
+          paddingLeft: 10,
+          paddingRight: 10,
+          backgroundColor: "white",
+          display: "flex",
+          justifyContent: "space-between"
+          // boxShadow: "0px -3px 6px 2px rgba(153,153,153,0.4)",
+          // borderBottom: "2px solid red"9
+        }}
+      >
+        <div style={{ flexShrink: 0 }}>
+          <Logo size={30} />
+          <span
+            style={{
+              fontSize: 28,
+              position: "relative",
+              top: -6,
+              fontWeight: "bold",
+              marginLeft: 5,
+              color: globals.logoColor,
+              userSelect: "none"
+            }}
+          >
+            cell<span
+              style={{
+                position: "relative",
+                top: 1,
+                fontWeight: 300,
+                fontSize: 24
+              }}
+            >
+              ×
+            </span>gene
+          </span>
+          <span
+            data-testid="header"
+            style={{
+              fontSize: 14,
+              position: "relative",
+              marginLeft: 7,
+              top: -8
+            }}
+          >
+            {datasetTitle}
+          </span>
+        </div>
+        <div
+          style={{
+            marginRight: 10,
+            marginBottom: 10,
+            flexShrink: 0
+          }}
+        >
+          <CellSetButton {...this.props} eitherCellSetOneOrTwo={1} />
+          <CellSetButton {...this.props} eitherCellSetOneOrTwo={2} />
+          {!differential.diffExp ? (
+            <Tooltip
+              content="Add two cells selections, see the top 15 differentially expressed genes between them"
+              position="bottom"
+            >
+              <AnchorButton
+                disabled={!haveBothCellSets}
+                intent="primary"
+                data-testid="diffexp-button"
+                loading={differential.loading}
+                fill
+                type="button"
+                onClick={this.computeDiffExp.bind(this)}
```

Comment:
if you define these functions as a class variable, they will be already bound to `this`, and you can get rid of the explicit bind (and make the linter happy).

In other words, define them as:
```
computeDiffExp = () => { ... }
```

Reply:
there is an example in `isClipDisabled()`